Plugins/Object/Object.cpp

```diff
@@ -820,7 +821,67 @@ NWNX_EXPORT ArgumentStack DoSpellLevelAbsorption(ArgumentStack&& args)
     if (auto *pObject = Utils::PopObject(args))
     {
         if(auto *pVersus = Utils::PopObject(args))
-            return pObject->DoSpellLevelAbsorption(pVersus);
+        {
+            const auto spellId     = args.extract<int32_t>();
+            const auto spellLevel  = args.extract<int32_t>();
+              ASSERT_OR_THROW(spellLevel <= 10);
+            const auto spellSchool = args.extract<int32_t>();
+              ASSERT_OR_THROW(spellSchool <= 8);
```

Comment:
If you do something like this:
```c
int32_t spellId, spellLevel, spellSchool;        
try
{
    spellId = args.extract<int32_t>();
    spellLevel = args.extract<int32_t>();
    spellSchool = args.extract<int32_t>();
}
catch(const std::runtime_error& e)
{
    LOG_WARNING("NWNX_Object_DoSpellLevelAbsorption() called from NWScript without new parameters. Please update nwnx_object.nss");
    spellId = -1;
    spellLevel = -1;
    spellSchool = -1;
}

ASSERT_OR_THROW(spellLevel <= 10);
ASSERT_OR_THROW(spellSchool <= 8);
```

it won't break the ABI and previously compiled scripts.

Reply:
Nice! didn't know we can do this.